recipes/crashpad/all/conanfile.py

```diff
@@ -191,6 +191,14 @@ def build(self):
                     targets=" ".join(targets),
                     parallel=tools.cpu_count()), run_environment=True)
 
+        def lib_filename(name):
+            prefix, suffix = ("", ".lib")  if self.settings.compiler == "Visual Studio" else ("lib", ".a")
+            return "{}{}{}".format(prefix, name, suffix)
+        tools.rename(os.path.join(self._source_subfolder, "out", "Default", "obj", "client", lib_filename("common")),
+                     os.path.join(self._source_subfolder, "out", "Default", "obj", "client", lib_filename("client_common")))
+        tools.rename(os.path.join(self._source_subfolder, "out", "Default", "obj", "handler", lib_filename("common")),
+                     os.path.join(self._source_subfolder, "out", "Default", "obj", "handler", lib_filename("handler_common")))
```

Comment:
Wouldn't the correct thing be to have them in `<pkg>/lib/client/common.lib` & `<pkg>/lib/handler/common.lib` respectively (or whatever is closest to the way the library outputs them)? I think it's important to avoid changing names of output files when packaging them, because otherwise that makes the conan package less of a "drop-in replacement", eg. it can't be use without "do this special handling since it's the conan package".

The same goes for `crashpad_handler_com.com`, it should be named `crashpad_handler.com` since this is what the library does. Otherwise, again, consumers would have to know about these package "quirks" when using conan (I for example always use cmake `find_package(...)` and expect things to work the same regardless of how I got the library).

Reply:
> Wouldn't the correct thing be to have them in `<pkg>/lib/client/common.lib` & `<pkg>/lib/handler/common.lib` respectively (or whatever is closest to the way the library outputs them)? I think it's important to avoid changing names of output files when packaging them, because otherwise that makes the conan package less of a "drop-in replacement", eg. it can't be use without "do this special handling since it's the conan package".

I don't know whether this would work:
```
self.cpp_info.components["handler_common"].libs = ["common"]
self.cpp_info.components["handler_common"].libdirs = os.path.join("lib", "handler")

self.cpp_info.components["client_common"].libs = ["common"]
self.cpp_info.components["client_common"].libdirs = os.path.join("lib", "client")
```

It is unclear what library will be chosen when using `-lcommon`.
Naming a library common is very prone to collisions anyway.
Sentry's crashpad fork prepends all libraries with `crashpad_`.
`mini_chromium` creates a base library, which will/might cause problems in the future too.

(I would prepend all libraries with `crashpad_`)

Reply:
> Or maybe it is named `crashpad_handler_com.com`? Documentation is a bit fuzzy on this one.
> **Edit**
> Nah `crashpad_handler.com` is the correct name.

I renamed it to `crashpad_handler.com`.

Reply:
I renamed it to `crashpad_handler.com`.
Docs say:
> On Windows, this executable is built by default as a Windows GUI app, so no
console will appear in normal usage. This is the version that will typically be
used. A second copy is also made with a `.com` extension, rather than `.exe`. In
this second copy, the PE header is modified to indicate that it’s a console app.
This is useful because the `.com` is found in the path before the `.exe`, so
when run normally from a shell using only the basename (without an explicit
`.com` or `.exe` extension), the `.com` console version will be chosen, and so
stdio will be hooked up as expected to the parent console so that logging output
will be visible.

No words on the `_com`.

Reply:
> > Wouldn't the correct thing be to have them in `<pkg>/lib/client/common.lib` & `<pkg>/lib/handler/common.lib` respectively (or whatever is closest to the way the library outputs them)? I think it's important to avoid changing names of output files when packaging them, because otherwise that makes the conan package less of a "drop-in replacement", eg. it can't be use without "do this special handling since it's the conan package".
> 
> I don't know whether this would work:
> 
> ```
> self.cpp_info.components["handler_common"].libs = ["common"]
> self.cpp_info.components["handler_common"].libdirs = os.path.join("lib", "handler")
> 
> self.cpp_info.components["client_common"].libs = ["common"]
> self.cpp_info.components["client_common"].libdirs = os.path.join("lib", "client")
> ```

Doesn't this work: `self.cpp_info.components["client_common"].libs = ["client/common"]` ? 
(I'm guessing here. I don't know enough about conan components).

That way the (conan) component can be called `client_common` but the actual file is still called just `common.lib`.

Reply:
> Doesn't this work: `self.cpp_info.components["client_common"].libs = ["client/common"]` ?
> (I'm guessing here. I don't know enough about conan components).
> 
> That way the (conan) component can be called `client_common` but the actual file is still called just `common.lib`.

Perhaps. I don't know either.
But I'ld prefer to not try out anything special here.
I think the `common` libraries are "internal" libraries of crashpad and should be treated as implementation detail.
Renaming the libraries is a good middle ground, for now.
We'll come back to this if some dependency really needs a library named `common`.
I highly doubt such a dependency exists. And if it does, then its build system is really fragile and broken already.